packages/flutter/lib/src/widgets/inherited_theme.dart

```diff
@@ -6,6 +6,38 @@
 library;
 
 import 'framework.dart';
+import 'inherited_model.dart';
+
+/// A generic selector interface for theme data.
+///
+/// This interface allows for selecting specific aspects of theme data without
+/// coupling to specific theme implementations like Material or Cupertino.
+///
+/// The type parameter [T] represents the type of the theme data object
+/// (e.g., `ThemeData`, `CupertinoThemeData`).
+/// The type parameter [V] represents the type of the value being selected
+/// from the theme data (e.g., `Color`, `TextStyle`).
+abstract interface class ThemeSelector<T, V> {
+  /// Creates a [ThemeSelector] from a function.
+  ///
+  /// This factory constructor allows for creating a selector using a lambda syntax.
+  factory ThemeSelector.from(V Function(T) selector) = _FunctionThemeSelector<T, V>;
```

Comment:
It looks like each look up will create a different aspect? and the updateShouldNotifyDependent runtime will also grow with number of aspects. Not sure how bad this will be.

Reply:
every rebuild may also stack the aspect on top

Reply:
Yes, aspect and updateShouldNotifyDependent will be linearly growing with each lookup.
>every rebuild may also stack the aspect on top
I think that will be resolved with equal and hashcode implementation.

Reply:
> equal and hashcode

It only helps for those who are careful to use private method as getter.

so only this case will be fine
```dart
double _myGetter(IconTheme theme) => theme.size;

IconTheme.selectFrom(
      context,
      _myGetter
    );
```

If they go lazy and use inline closure, the equality won't work
```dart
IconTheme.selectFrom(
      context,
      (IconTheme theme) => theme.size,
    );
```

Reply:
And many times, they will go for inline closure only.

Reply:
@chunhtai For this specific case, can we use selector.toString().hashCode for equality ?